web/html/src/manager/maintenance/calendar/web-calendar.tsx

```diff
@@ -0,0 +1,262 @@
+          const lastEvent = moment.parseZone(filteredEvents[filteredEvents.length - 1].start);
+          getApi().gotoDate(lastEvent.format("YYYY-MM-DD"));
+          props.clearMessages();
+          setCurrentDate(getApi().currentDataManager.data.currentDate);
+        }
+      }).catch(props.responseError)
+  };
+
+  const onSkipNext = () => {
+    // Get last date of currently displayed range
+    let date = getDate("next");
+    // Check if we need to update the events
+    if (!needsUpdate(date) && skipToNextEvent(date)) {
+      return;
+    }
+    // If we are in day view and need to update we want to make sure to always use the first day of the next month
+    if (getApi().currentDataManager.data.currentViewType === "timeGridDay" && moment(date).date() !== 1) {
+      date = Date.parse(moment(date).add(1, 'month').startOf('month'));
+    }
+    const endpoint = `/rhn/manager/api/maintenance/events/skipNext/${props.type}/${date}/${props.id}`;
+    return Network.get(endpoint, "application/json").promise
+      .then(events => {
+        if (events.data.length === 0) {
+          props.messages(MessagesUtils.info(t("There are no more future maintenance windows")));
+        } else {
+          setEvents(events.data);
+          // Skip to next event that is not in current month
+          const firstEvent = moment.parseZone(events.data.filter(event =>
+            moment.parseZone(event.start).month() !== moment(currentDate).month())[0].start
+          );
+          getApi().gotoDate(firstEvent.format("YYYY-MM-DD"));
+          props.clearMessages();
+          setCurrentDate(getApi().currentDataManager.data.currentDate);
+        }
+      }).catch(props.responseError)
+  };
+
+  const onClickMonth = () => {
+    if (getApi().currentDataManager.data.currentViewType === "dayGridMonth") {
+      return;
+    }
+    if (needsUpdate(getApi().currentDataManager.data.currentDate)) {
+      getEvents("current");
+    }
+    getApi().changeView("dayGridMonth", getApi().currentDataManager.data.currentDate);
+  };
+
+  const onClickDay = () => {
+    getApi().changeView("timeGridDay", getApi().currentDataManager.data.currentDate);
+  };
+
+  // Switch to the day view of the clicked date
+  const onDateClick = (info) => {
+    setCurrentDate(getApi().currentDataManager.data.currentDate);
+    getApi().changeView('timeGridDay', info.date);
+    if (needsUpdate(info.date)) {
+      getEvents("current");
+    }
+  };
+
+  // Get the date used to fetch events based on the operation
+  const getDate = (operation) => {
+    switch (operation) {
+      case "next":
+        return Date.parse(getApi().currentDataManager.state.dateProfile.currentRange.end);
+      case "back":
+        // currentRange.start returns the first day of the displayed month, but we want to end up in
+        // the previous month so we subtract one day in this case
+        return Date.parse(getApi().currentDataManager.state.dateProfile.currentRange.start) - 24 * 60 * 60 * 1000;
```

Comment:
Here and elsewhere, while this is not wrong per se, it will be easier to reason about and change if need be if we use moment to modify the underlying object as is already done in `onSkipNext `.

Reply:
I have no preference here. This is just a leftover from before I used moment to format the dates. I can change it to be in line with the rest.